engine/order_manager_test.go

```diff
@@ -559,3 +559,44 @@ func TestProcessOrders(t *testing.T) {
 	m := OrdersSetup(t)
 	m.processOrders()
 }
+
+func TestGetOrdersFiltered(t *testing.T) {
+	m := OrdersSetup(t)
+	_, err := m.GetOrdersFiltered(nil)
+	if err == nil {
+		t.Error("Expected error from nil filter")
+	}
+	// main functionality testing is done in Test_getFilteredOrders
```

Comment:
I understand that the testing of `getFilteredOrders` is done in more detail with the test `Test_getFilteredOrders`, but it would be good to test each type of scenario for `GetOrdersFiltered` as this test is only checking for one type of failure. We do not care for the result of `getFilteredOrders` here, but knowing via testing that you can actually call the function successfully would improve this test a lot